src/Magento/FunctionalTestingFramework/Console/RunTestCommand.php

```diff
@@ -76,60 +81,82 @@ protected function execute(InputInterface $input, OutputInterface $output): int
-            if (!realpath($testsDirectory . $testGroup . $testName)) {
+        $testsDirectory = TESTS_MODULE_PATH .
+            DIRECTORY_SEPARATOR .
+            TestGenerator::GENERATED_DIR .
+            DIRECTORY_SEPARATOR .
+            TestGenerator::DEFAULT_DIR .
+            DIRECTORY_SEPARATOR ;
+
+        foreach ($tests as $test) {
+            $testName = $test . 'Cest.php';
+            if (!realpath($testsDirectory . $testName)) {
                 throw new TestFrameworkException(
-                    $testName . " is not available under " . $testsDirectory . $testGroup
+                    $testName . " is not available under " . $testsDirectory
                 );
             }
-            $fullCommand = $codeceptionCommand . $testsDirectory . $testGroup . $testName . ' --verbose --steps';
+            $fullCommand = $codeceptionCommand . $testsDirectory . $testName . ' --verbose --steps';
             $process = new Process($fullCommand);
             $process->setWorkingDirectory(TESTS_BP);
             $process->setIdleTimeout(600);
             $process->setTimeout(0);
-
-            $returnCode = max($returnCode, $process->run(
-                function ($type, $buffer) use ($output) {
-                    $output->write($buffer);
-                }
-            ));
+            $subReturnCode = $process->run(function ($type, $buffer) use ($output) {
+                $output->write($buffer);
+            });
+            $this->returnCode = max($this->returnCode, $subReturnCode);
         }
-        return $returnCode;
     }
 
     /**
-     * Get an array of tests with resolved suite references from $testConfiguration
-     * eg: if test is referenced in a suite, it'll be stored in format suite:test
-     * @param string $testConfigurationJson
-     * @return array
+     * Run tests referenced in suites within suites' context.
+     *
+     * @param array           $suitesConfig
+     * @param OutputInterface $output
+     * @return void
+     *
+     * @SuppressWarnings(PHPMD.UnusedLocalVariable)
      */
-    private function resolveSuiteReferences($testConfigurationJson)
+    private function runTestsInSuite(array $suitesConfig, OutputInterface $output)
     {
-        $testConfiguration = json_decode($testConfigurationJson, true);
-        $testsArray = $testConfiguration['tests'] ?? [];
-        $suitesArray = $testConfiguration['suites'] ?? [];
-        $testArrayBuilder = [];
-
-        foreach ($suitesArray as $suite => $tests) {
-            foreach ($tests as $test) {
-                $testArrayBuilder[] = "$suite:$test";
-            }
+        $codeceptionCommand = realpath(PROJECT_ROOT . '/vendor/bin/codecept') . ' run functional --verbose --steps ';
+        //for tests in suites, run them as a group to run before and after block
+        foreach (array_keys($suitesConfig) as $suite) {
+            $fullCommand = $codeceptionCommand . " -g {$suite}";
+            $process = new Process($fullCommand);
```

Comment:
Ln 152 - 158 are duplicates of 126-132. Abstract it out into its own function so that you can just call something like
```
$subreturnCode = $this->executeTestCommand($fullCommand);
$this->returnCode = max($this->returnCode, $subreturnCode);
```
Double check for any more duped variable assignments/etc for consolidation as well.

Reply:
created a new function executeTestCommand that runs the codecept command.